### PR TITLE
ci: add version check on release branches

### PR DESCRIPTION
**Pull request by andrewjl-mux**
This check only handles Cocoapods at the moment, checking Version.swift requires additional tooling and will be added in a separate pull request.

Here's an example of a detected version mismatch on a mock release branch: https://github.com/muxinc/swift-upload-sdk/actions/runs/5721374427/job/15502867962

### CODE REVIEW

File: scripts/version-check.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+echo "Detected Cocoapod Spec Version: ${cocoapod_spec_version}"
+
+release_version=$(git branch --show-current | sed -E 's/.*v([0-9]+\.[0-9]+\.[0-9]+).*/\1/')
+
+echo "Inferred Release Version: ${release_version}"
+
+if [ "${cocoapod_spec_version}" == "${release_version}" ]; then
+	echo "Versions match"
+else
+    echo "Versions do not match, please update the cocoapod spec to ${release_version}"
```
Comment (cjpillsbury) [inline scripts/version-check.sh:16]: suggestion(non-blocking):

```suggestion
    echo "Versions do not match, please update the cocoapod spec ${COCOAPOD_SPEC} to ${release_version}"
```

File: .github/workflows/validate-release.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  versioncheck:
+    runs-on: macos-latest
+    name: Validate Versions
+    steps:
+      - name: Compare Podspec Version
+        uses: actions/checkout@v2
+      - name: Run Version Check Script
+        run: ./scripts/version-check.sh
+
```
Comment (cjpillsbury) [inline .github/workflows/validate-release.yml:17]: suggestion(non-blocking): Add newline at EOF

**Review by cjpillsbury — APPROVED** [review]
Love it! Approved with a couple of nb suggestions

File: .github/workflows/validate-release.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: Validate release
+
+on:
+  push:
+    branches:
+      - 'releases/**'
```
Comment (cjpillsbury) [inline .github/workflows/validate-release.yml:6]: question(non-blocking): This is targeting `main`. Should we also run this check against that branch?

Comment (andrewjl-mux) [reply .github/workflows/validate-release.yml:6]: One of the two versions being compared comes from the version being embedded in the branch name as in `releases/vX.Y.Z`, so we'd need to compare something else if running on `main`.

Comment (daytime-em) [reply .github/workflows/validate-release.yml:6]: Echoing that this kind of logic feels useful for releasing, but less useful at any other time. `main` may not point to the latest version if we change something that isn't versioned (like adding a line to the README, or fixing a crash in the example app), and dev/topic branches should be scoped below things versioning

Comment (cjpillsbury) [reply .github/workflows/validate-release.yml:6]: Yup was just thinking that we could gate merges to main to make sure we don't unintentionally merge an older version in. Probably overkill though, and def more complicated/would require pinning down what is allowed in `main` based on e.g. commit history and existing release branches, per both of your callouts.

File: scripts/version-check.sh (hunk 1 above)
Comment (andrewjl-mux) [reply scripts/version-check.sh]: Good suggestion!

**Pull request merged**